tests/docker_utils.py

```
@@ -0,0 +1,116 @@
+if TYPE_CHECKING:
+    from docker.models.containers import Container
+
+
+def container_cp(container: Container, src: str, dst: str,
+                 user: int | None = None, group: int | None = None) -> None:
+    """
+    Copies a directory into a container
+    From https://stackoverflow.com/questions/46390309/how-to-copy-a-file-from-host-to-container-using-docker-py-docker-sdk
+    """
+    fh = io.BytesIO()
+    tar = tarfile.open(fileobj=fh, mode="w:gz")
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)